File: app/src/main/java/io/recebimento/network/BrasilRiskClient.kt
package io.recebimento.network

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query
import java.util.concurrent.TimeUnit

interface BrasilRiskService {

    // ========== LOGIN BRLOG (token Microsoft) ==========
    @POST("validar/LoginMobileDeliveryClientMicrosoft")
    fun loginMicrosoft(
        @Body body: TokenBody
    ): Call<BrasilRiskLoginResponse>

    // ========== DETALHE DA NOTA (rastreio BRLog) ==========
    @GET("listar/ObterInformacoesNotaCliente")
    fun obterInformacoesNotaCliente(
        @Query("CodEmpresaUsuario") codEmpresaUsuario: Int?,
        @Query("CodPedido") codPedido: Int?
    ): Call<BrasilRiskNotaDetalhe>

    // ========== AUTORIZAR ABERTURA DO BAÚ (liberação) ==========
    @POST("salvar/AutorizarAberturaBau")
    fun autorizarAberturaBau(
        @Body body: AutorizarAberturaBauRequest
    ): Call<BrasilRiskBauResponse>
}

object BrasilRiskClient {

    private const val BASE_URL = "https://apimobile.brasilrisk.com.br/"

    private val okHttpClient by lazy {
        OkHttpClient.Builder()
            .addInterceptor(
                HttpLoggingInterceptor().apply {
                    level = HttpLoggingInterceptor.Level.BASIC
                }
            )
            .connectTimeout(30, TimeUnit.SECONDS)
            .readTimeout(30, TimeUnit.SECONDS)
            .writeTimeout(30, TimeUnit.SECONDS)
            .build()
    }

    private val retrofit by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .client(okHttpClient)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    fun getService(): BrasilRiskService = retrofit.create(BrasilRiskService::class.java)
}

File: flutter/lib/network/brasil_risk_client.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../../utils/constants.dart';
import '../../models/brasil_risk.dart';

class BrasilRiskClient {
  static const String baseUrl = Constants.brasilRiskBaseUrl;

  Future<BrasilRiskLoginResponse> loginMicrosoft(TokenBody body) async {
    final uri = Uri.parse(baseUrl + 'validar/LoginMobileDeliveryClientMicrosoft');
    final response = await http.post(uri,
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode(body.toJson()));
    return BrasilRiskLoginResponse.fromJson(
        jsonDecode(response.body) as Map<String, dynamic>);
  }

  Future<BrasilRiskNotaDetalhe> obterInformacoesNotaCliente(
      int? codEmpresaUsuario, int? codPedido) async {
    final query = <String, String>{
      'CodEmpresaUsuario': (codEmpresaUsuario ?? 0).toString(),
      'CodPedido': (codPedido ?? 0).toString(),
    };
    final uri = Uri.parse(baseUrl + 'listar/ObterInformacoesNotaCliente')
        .replace(queryParameters: query);
    final response = await http.get(uri);
    return BrasilRiskNotaDetalhe.fromJson(
        jsonDecode(response.body) as Map<String, dynamic>);
  }

  Future<BrasilRiskBauResponse> autorizarAberturaBau(
      AutorizarAberturaBauRequest request) async {
    final uri = Uri.parse(baseUrl + 'salvar/AutorizarAberturaBau');
    final response = await http.post(uri,
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode(request.toJson()));
    return BrasilRiskBauResponse.fromJson(
        jsonDecode(response.body) as Map<String, dynamic>);
  }
}
